packages/flutter/lib/src/material/predictive_back_page_transitions_builder.dart

```diff
@@ -0,0 +1,325 @@
+  }
+
+  @override
+  bool handleUpdateBackGestureProgress(PredictiveBackEvent backEvent) {
+    if (!_gestureInProgress) {
+      return false;
+    }
+
+    widget.route.handleUpdateBackGestureProgress(progress: 1 - backEvent.progress);
+    currentBackEvent = backEvent;
+    return true;
+  }
+
+  @override
+  bool handleCancelBackGesture() {
+    if (!_gestureInProgress) {
+      return false;
+    }
+
+    widget.route.handleCancelBackGesture();
+    _gestureInProgress = false;
+    startBackEvent = currentBackEvent = null;
+    return true;
+  }
+
+  @override
+  bool handleCommitBackGesture() {
+    if (!_gestureInProgress) {
+      return false;
+    }
+
+    widget.route.handleCommitBackGesture();
+    _gestureInProgress = false;
+    startBackEvent = currentBackEvent = null;
+    return true;
+  }
+
+  // End WidgetsBindingObserver.
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addObserver(this);
+  }
+
+  @override
+  void dispose() {
+    WidgetsBinding.instance.removeObserver(this);
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return widget.builder(context);
+  }
+}
+
+/// Android's predictive back page transition.
+class _PredictiveBackPageTransition extends StatelessWidget {
+  const _PredictiveBackPageTransition({
+    required this.animation,
+    required this.secondaryAnimation,
+    required this.getIsCurrent,
+    required this.child,
+  });
+
+  // These values were eyeballed to match the native predictive back animation
+  // on a Pixel 2 running Android API 34.
+  static const double _scaleFullyOpened = 1.0;
+  static const double _scaleStartTransition = 0.95;
+  static const double _opacityFullyOpened = 1.0;
+  static const double _opacityStartTransition = 0.95;
+  static const double _weightForStartState = 65.0;
+  static const double _weightForEndState = 35.0;
+  static const double _screenWidthDivisionFactor = 20.0;
+  static const double _xShiftAdjustment = 8.0;
+
+  final Animation<double> animation;
+  final Animation<double> secondaryAnimation;
+  final ValueGetter<bool> getIsCurrent;
+  final Widget child;
+
+  Widget _secondaryAnimatedBuilder(BuildContext context, Widget? child) {
+    final Size size = MediaQuery.sizeOf(context);
+    final double screenWidth = size.width;
+    final double xShift =
+        (screenWidth / _screenWidthDivisionFactor) - _xShiftAdjustment;
+
+    final bool isCurrent = getIsCurrent();
+    final Tween<double> xShiftTween = isCurrent
+        ? ConstantTween<double>(0)
+        : Tween<double>(begin: xShift, end: 0);
+    final Animatable<double> scaleTween = isCurrent
+        ? ConstantTween<double>(_scaleFullyOpened)
+        : TweenSequence<double>(<TweenSequenceItem<double>>[
+            TweenSequenceItem<double>(
+                tween: Tween<double>(
+                    begin: _scaleStartTransition, end: _scaleFullyOpened),
+                weight: _weightForStartState),
```

Comment:
formatting nit here and elsewhere:

```suggestion
              tween: Tween<double>(
                begin: _scaleStartTransition, 
                end: _scaleFullyOpened,
              ),
              weight: _weightForStartState,
            ),
```